WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/lists/widget/ViewsWidgetUpdater.kt

```diff
@@ -5,69 +5,65 @@ import android.appwidget.AppWidgetManager
 import android.content.ComponentName
 import android.content.Context
 import android.content.Intent
-import android.net.Uri
-import android.view.View
-import android.widget.ImageView.ScaleType.FIT_START
 import android.widget.RemoteViews
-import com.bumptech.glide.request.target.AppWidgetTarget
 import org.wordpress.android.R
-import org.wordpress.android.WordPress
 import org.wordpress.android.fluxc.store.SiteStore
 import org.wordpress.android.ui.prefs.AppPrefsWrapper
-import org.wordpress.android.ui.stats.OldStatsActivity
-import org.wordpress.android.ui.stats.StatsTimeframe
+import org.wordpress.android.ui.stats.StatsTimeframe.DAY
 import org.wordpress.android.ui.stats.refresh.StatsActivity
-import org.wordpress.android.ui.stats.refresh.lists.widget.StatsWidgetConfigureFragment.ViewType
-import org.wordpress.android.ui.stats.refresh.lists.widget.StatsWidgetConfigureViewModel.Color
+import org.wordpress.android.ui.stats.refresh.lists.widget.StatsWidgetConfigureFragment.ViewType.WEEK_VIEWS
 import org.wordpress.android.ui.stats.refresh.lists.widget.StatsWidgetConfigureViewModel.Color.DARK
 import org.wordpress.android.ui.stats.refresh.lists.widget.StatsWidgetConfigureViewModel.Color.LIGHT
 import org.wordpress.android.util.NetworkUtilsWrapper
-import org.wordpress.android.util.image.ImageManager
-import org.wordpress.android.util.image.ImageType.ICON
 import org.wordpress.android.viewmodel.ResourceProvider
 import javax.inject.Inject
-import kotlin.random.Random
-
-private const val MIN_WIDTH = 250
 
 class ViewsWidgetUpdater
 @Inject constructor(
     private val appPrefsWrapper: AppPrefsWrapper,
     private val siteStore: SiteStore,
-    private val imageManager: ImageManager,
     private val networkUtilsWrapper: NetworkUtilsWrapper,
-    private val resourceProvider: ResourceProvider
+    private val resourceProvider: ResourceProvider,
+    private val widgetUtils: WidgetUtils
 ) : WidgetUpdater {
     override fun updateAppWidget(
         context: Context,
         appWidgetManager: AppWidgetManager,
         appWidgetId: Int
     ) {
-        val minWidth = appWidgetManager.getAppWidgetOptions(appWidgetId)
-                .getInt(AppWidgetManager.OPTION_APPWIDGET_MIN_WIDTH, 300)
-        val showChangeColumn = minWidth > MIN_WIDTH
+        val showChangeColumn = widgetUtils.isWidgetWiderThanLimit(appWidgetManager, appWidgetId)
         val colorMode = appPrefsWrapper.getAppWidgetColor(appWidgetId) ?: LIGHT
         val siteId = appPrefsWrapper.getAppWidgetSiteId(appWidgetId)
         val siteModel = siteStore.getSiteBySiteId(siteId)
         val networkAvailable = networkUtilsWrapper.isNetworkAvailable()
         val layout = when (colorMode) {
-            DARK -> R.layout.stats_widget_views_dark
-            LIGHT -> R.layout.stats_widget_views_light
+            DARK -> R.layout.stats_widget_list_dark
+            LIGHT -> R.layout.stats_widget_list_light
         }
         val views = RemoteViews(context.packageName, layout)
```

Comment:
Should we use `getLayout` from `WidgetUtils` here too?

Reply:
the `Views` widget doesn't show the block layout. We can create a second method (maybe `getListLayout`) which we can reuse in all the widgets